frontend/javascripts/oxalis/view/right-border-tabs/tree_hierarchy_view.tsx

```diff
@@ -254,11 +253,11 @@ function TreeHierarchyView(props: Props) {
   const checkedKeys = deepFlatFilter(UITreeData, (node) => node.isChecked).map((node) => node.key);
 
   // selectedKeys is mainly used for highlighting, i.e. blueish background color
-  const selectedKeys = props.selectedTreeIds.map((treeId) =>
-    getNodeKey(GroupTypeEnum.TREE, treeId),
-  );
+  let selectedKeys = props.selectedTreeIds.map((treeId) => getNodeKey(GroupTypeEnum.TREE, treeId));
+
+  if (props.activeGroupId) selectedKeys = [getNodeKey(GroupTypeEnum.GROUP, props.activeGroupId)];
```

Comment:
Is there a reason why you remove the selected keys here in case there is an active group?

Reply:
yes! it is because when selecting multiple elements with ctrl, you cannot select skeletons and skeleton groups at the same time. next to that, there is no multi group selection. Its not like there is a distiction between active and selected group like for the segments, right?

Reply:
There is even this modal which says "You have ${numberOfSelectedTrees} selected Trees. Do you really want to select this group? This will deselect all selected trees." 

I thought about adding this here to in this specific case, but it seems hard to realize without very much changing the current architecture:
- the logic that the selected key is a group is within tree_hierarchy_view (for the skeletons tab)
- we would need it in`AdvancedSearchPopover` though, because if the user selects "no" within the modal, we dont want to change the selection. As there is no history to the selection, this needs to happen in the search popover modal.
- the components dont know each other, they are both placed in the `SkeletonView`

Adding to this, the user pressed the arrow to select the next tree / treegroup, so its hopefully not surprising behaviour. So I personally dont see the need to restructure the code for this to work.
What do you think?

Reply:
Oh right, I forgot about this limitation. 
> There is even this modal which says "You have ${numberOfSelectedTrees} selected Trees. Do you really want to select this group? This will deselect all selected trees."

Thanks for the clarification :pray: 
So you kinda fixed the code a little here, correct? It behaves "more correctly" than before. I guess the code before was not broken because when a group was selected the code took care of deselecting all trees? Beside that, in regard to your clarification your changes are much nicer than the previous version :sparkles: 

> So I personally dont see the need to restructure the code for this to work.
What do you think?

Sure, fine by me. If we get the feedback that this should be included in the "search multi selection" then we could still develop this as a follow up :)

Reply:
Yes, I think I removed inconsistent behaviour regarding the multiselection via ctrl vs the selection in the search. Previously if you had selected skeletons and selected a group via the search, nothing happened to the skeleton selection. I think this wasnt too bad but it doesnt make much sense either. you cannot do any actions with your heterogenous selection anyway, such as drag and drop or actions via a context menu
![image](https://github.com/user-attachments/assets/7c50cf98-5f90-430d-a9b2-fe3fc208b891)